text/1003-deprecation-import-ember-from-ember.md

```diff
@@ -0,0 +1,416 @@
+prs:
+  accepted: Fill this in with the URL for the Proposal RFC PR
+project-link: Leave as is
+-->
+
+# Deprecate `import Ember from 'ember'; 
+
+## Summary
+
+This RFC proprosing deprecating all APIs that have module-based replacements, as described in [RFC #176](https://rfcs.emberjs.com/id/0176-javascript-module-api) as well as other `Ember.*` apis that are no longer needed.
+
+## Motivation
+
+The `import Ember from 'ember';` set of APIs is implementn as a barrel file, and properly optimizing barrel files [is a lot of work, requiring integration with build time tools](https://vercel.com/blog/how-we-optimized-package-imports-in-next-js).
+
+**If anyone one dependency in an app's dependency tree does `import ... from 'ember'`, every feature of the framework is shipped to your users, without any ability for you to optimize.**
+
+By removing this set of exports, we have an opportunity to shrink some apps (as some APIs are not used), improving the load performance of ember apps -- and we removing all of these gives us a chance to have a better grasp of what we can get rid of permananently.
+
+Many of these APIs already have alternatives, and those will be called out explicitly in the _Transition Path_ below.
+
+## Transition Path
+
+This list is semi-exhaustive, in that it covers _every_ export from 'ember', but may not exhaustivily provide alternatives.
+
+Throughout the rest of this RFC, the following key will be used:
+- 🌐 to mean "this is public API"
+- 🔒 to mean "this is private API"
+- 🧷 to mean "this is protected API"
+- 🫣 to mean "no declared access"
+
+### Testing utilities
+
+APIs for wiring up a test framework (e.g. QUnit, _etc_)
+
+|   | API | Usage: EmberObserver | -- | 
+| - | --- | ----- | -- |
+|🌐 | `Ember.Test` | [A good few](https://emberobserver.com/code-search?codeQuery=Ember.Test&sort=updated&sortAscending=false) | |
+|🌐 | `Ember.Test.Adapter` | [`ember-cli-fastboot` and `@ember/test-helpers`](https://emberobserver.com/code-search?codeQuery=Ember.Test.Adapter&sort=updated&sortAscending=false) | currently available at [`@ember/test`](https://api.emberjs.com/ember/5.6/modules/@ember%2Ftest) |
+|🌐 | `Ember.Test.QUnitAdapter` | [`ember-cli-fastboot`](https://emberobserver.com/code-search?codeQuery=Ember.Test.QUnitAdapter&sort=updated&sortAscending=false) | |
+|🌐 | `Ember.setupForTesting` | [`ember-cli-fastboot`](https://emberobserver.com/code-search?codeQuery=Ember.setupForTesting&sort=updated&sortAscending=false) | | 
+
+If needed, these will need to be moved to a module such as `@ember/test`.
+
+### A way to communicate with the ember-inspector
+
+The inspector will be hit especially hard by the removal of these APIs.
+
+A good few already have available imports though.
+
+|   | API | import |
+| - | --- | ------ |
+|🔒| `Ember.meta` | `import { meta } from '@ember/-internals/meta';` |
+|🌐| `Ember.VERSION` | none, we should add one, `@ember/version` |
```

Comment:
This seems to exist as `import { VERSION } from '@ember/version'`.

Reply:
updated, thanks!